cpp/models/abm/person.cpp

```diff
@@ -234,36 +240,30 @@ ScalarType Person::get_mask_protective_factor(const Parameters& params) const
     }
 }
 
-bool Person::apply_mask_intervention(RandomNumberGenerator& rng, const Location& target)
+bool Person::is_compliant(RandomNumberGenerator& rng, InterventionType intervention, const Location* target)
 {
-    if (target.get_npi_active() == false) {
-        m_wears_mask = false;
-        if (get_mask_compliance(target.get_type()) > 0.) {
-            // draw if the person wears a mask even if not required
-            ScalarType wear_mask = UniformDistribution<double>::get_instance()(rng);
-            if (wear_mask < get_mask_compliance(target.get_type())) {
-                m_wears_mask = true;
+    ScalarType compliance_check = UniformDistribution<double>::get_instance()(rng);
+    if (compliance_check <= get_compliance(intervention)) {
+        switch (intervention) {
+        case InterventionType::Mask:
+            m_wears_mask = true;
+            if (target != nullptr &&
+                static_cast<int>(m_mask.get_type()) < static_cast<int>(target->get_required_mask())) {
+                m_mask.change_mask(target->get_required_mask());
             }
+            break;
+        default:
+            // For now, there no Additional logic for testing and isolation intervention
+            break;
         }
+        return true;
```

Comment:
Thanks for the comments. I fixed them with the new code:

- The `is_compliant` is only dependent on the Person, hence to pointer to Location
- I compare directly the protection level using the World's parameters.
- The `is_compliant` and `set_mask(MaskType)` are separated and I also added the `MaskType None = 0` in the code.

Reply:
> I compare directly the protection level using the World's parameters.

That is a great idea, I like that a lot more than the enum or "mask level".
In World::migration, it confuses me a bit that a Person changes mask and instantly increases its time used. Is there maybe an else missing on the increase_time_used? Alternatively, we could save the time at which the mask was put on instead of the duration to avoid this?

Reply:
Thanks for the suggestion. I switched it to a TimePoint of the Mask's initial usage.